loki/transformations/temporaries/stack_allocator.py

```diff
@@ -0,0 +1,1372 @@
+                        expr_map = {
+                            expr: DetachScopesMapper()(arg_map[expr])\
+                                    for expr in FindVariables().visit(successor_stack_size)
+                            if expr in arg_map
+                        }
+                        if expr_map:
+                            expr_map = recursive_expression_map_update(expr_map)
+                            successor_stack_size = SubstituteExpressions(expr_map).visit(successor_stack_size)
+                        stack_dict.setdefault(dtype, {})
+                        stack_dict[dtype].setdefault(kind, []).append(successor_stack_size)
+
+
+        if not stack_dict:
+            # Return only the local stack size if there are no callees
+            return local_stack_dict or {}
+
+        # Unwind "max" expressions from successors and inject the local stack size into the expressions
+        for (dtype, kind_dict) in stack_dict.items():
+            for (kind, stack_sizes) in kind_dict.items():
+                new_list = []
+                for stack_size in stack_sizes:
+                    if (isinstance(stack_size, InlineCall) and stack_size.function == 'MAX'):
+                        new_list += list(stack_size.parameters)
+                    else:
+                        new_list += [stack_size]
+                stack_sizes = new_list
+
+        # simplify the local stack sizes and add them to the stack_dict
+        if local_stack_dict:
+            for dtype in local_stack_dict:
+                for kind in local_stack_dict[dtype]:
+                    local_stack_dict[dtype][kind] = DetachScopesMapper()(simplify(local_stack_dict[dtype][kind]))
+
+                    if dtype in stack_dict:
+                        if kind in stack_dict[dtype]:
+                            stack_dict[dtype][kind] = [simplify(Sum((local_stack_dict[dtype][kind], s)))
+                                                       for s in stack_dict[dtype][kind]]
+                        else:
+                            stack_dict[dtype][kind] = [local_stack_dict[dtype][kind]]
+                    else:
+                        stack_dict[dtype] = {kind: [local_stack_dict[dtype][kind]]}
+
+        # if several expressions, return MAX, else just add the expression
+        for (dtype, kind_dict) in stack_dict.items():
+            for (kind, stacks) in kind_dict.items():
+                if len(stacks) == 1:
+                    kind_dict[kind] = stacks[0]
+                else:
+                    kind_dict[kind] = InlineCall(function = Variable(name = 'MAX'), parameters = as_tuple(stacks))
+
+        return stack_dict
+
+class FtrPtrStackTransformation(BaseStackTransformation):
+    """         
+    Transformation to inject a stack that allocates large scratch spaces per block
+    and per datatype on the driver and maps temporary arrays in kernels to this scratch space.
+
+    Starting from:
+
+    .. code-block:: fortran
```

Comment:
This doesn't seem to render correctly, maybe missing empty lines or something (same in the other docstring): https://sites.ecmwf.int/docs/loki/554/loki.transformations.temporaries.stack_allocator.html#module-loki.transformations.temporaries.stack_allocator

I suggest to check the output of the documentation builder CI run for any warnings.